### PR TITLE
Zero copy

### PR DESCRIPTION
**Pull request by Kitrinx**
This PR adds options to accept hardware memory frame support for directx rendering.

### CODE REVIEW

**Review by Kodikuu — APPROVED** [review]
Ignoring the weird diff parts that I'm guessing were needed in order to build; Looks good

File: src/windows/gfx/d3d11.c [hunk 1]
```diff
@@ -301,6 +302,134 @@ static bool d3d11_refresh_resource(struct gfx *gfx, MTY_Device *_device, MTY_Con
...
+	if (plane > 0)
+		return true;
+
+	struct d3d11 *ctx = (struct d3d11 *) gfx;
+	bool r = true;
+
+	HANDLE shared_handle[3] = {NULL, NULL, NULL};
+	memcpy(&shared_handle[0], image, sizeof(HANDLE));
+	memcpy(&shared_handle[1], image + sizeof(HANDLE), sizeof(HANDLE));
+	memcpy(&shared_handle[2], image + (sizeof(HANDLE) * 2), sizeof(HANDLE));
```
Comment (RonaldH-Parsec) [inline src/windows/gfx/d3d11.c]: Is there a use for this to be in separate shared_handles?
Could this become:
```c
	HANDLE *shared_handle = (HANDLE *) image;
```

File: src/windows/gfx/d3d11.c [hunk 2]
```diff
@@ -301,6 +302,134 @@ static bool d3d11_refresh_resource(struct gfx *gfx, MTY_Device *_device, MTY_Con
...
+static bool d3d11_map_shared_resource(struct gfx *gfx, MTY_Device *_device, MTY_Context *context, MTY_ColorFormat fmt,
+	uint8_t plane, const uint8_t *image, uint32_t full_w, uint32_t w, uint32_t h, uint8_t bpp)
+{
+
+	// We do all textures at once so we dont want to deal with any later planes.
+	if (plane > 0)
+		return true;
+
+	struct d3d11 *ctx = (struct d3d11 *) gfx;
+	bool r = true;
```
Comment (RonaldH-Parsec) [inline src/windows/gfx/d3d11.c]: This one can be removed, since after the for loop, it will return true, and all except cases will return false.

**Review by RonaldH-Parsec — APPROVED** [review]

**Comment by RonaldH-Parsec** [comment]
Moved over to #112 because of signing issue in this one.

**Pull request closed without merging**